src/rnode-sign.js

```diff
@@ -1,13 +1,45 @@
+// @ts-check
 import blake from 'blakejs'
 import { ec } from 'elliptic'
 import jspb from 'google-protobuf'
 
+/**
+ * These deploy types are based on protobuf specification which must be
+ * used to create the hash and signature of deploy data.
+ * Deploy object sent to Web API is slightly different, see [rnode-web.js](rnode-web.js).
```

Comment:
very nice. I spent a lot of time puzzling over these types, looking at scala code, etc.

See also https://github.com/rchain-community/liquid-democracy/blob/dev-tools/rclient/types.js WIP